devimint/src/federation.rs

```diff
@@ -764,6 +766,80 @@ impl Federation {
         Ok(())
     }
 
+    pub async fn pegout_gateways(
```

Comment:
Can you document what this is doing, and why it performs multiple peg-outs instead of being called separately for each gateway? (I assume this is to prevent unnecessary block mining)